### PR TITLE
Fix bug related to augmenting client call details metatadata

### PR DESCRIPTION
**Pull request by BenRKarl**
Addresses this issue: https://github.com/googleads/google-ads-python/issues/16

Implementation based [on this](https://github.com/grpc/grpc/blob/master/examples/python/interceptors/headers/header_manipulator_client_interceptor.py) example from grpc.

### CODE REVIEW

File: google/ads/google_ads/client.py [hunk 1]
```diff
@@ -346,6 +347,21 @@ def __init__(self, developer_token, login_customer_id):
             ('login-customer-id', login_customer_id) if login_customer_id
             else None)
 
+    def _update_client_call_details_metadata(
+            self, client_call_details, metadata):
+        class _ClientCallDetails(
```
Comment (ericgribkoff) [inline google/ads/google_ads/client.py]: Google style guide recommends avoiding inner classes - I would move this definition outside of `class MetadataInterceptor`

Comment (BenRKarl) [reply google/ads/google_ads/client.py]: Awesome, that's great feedback. That would also make it easier to access in the tests if we decided to do so.

File: google/ads/google_ads/client.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import logging
 import os
 import yaml
+import collections
```
Comment (wihl) [inline google/ads/google_ads/client.py]: suggest `from collections import namedtuple`

File: google/ads/google_ads/client.py [hunk 3]
```diff
@@ -361,14 +370,22 @@ def intercept_unary_unary(self, continuation, client_call_details, request):
...
-        )
+        client_call_details = self._update_client_call_details_metadata(
+            client_call_details,
+            metadata)
 
         return continuation(client_call_details, request)
 
 
+class _ClientCallDetails(
+        collections.namedtuple(
```
Comment (wihl) [inline google/ads/google_ads/client.py]: then this would be `namedtuple`

File: google/ads/google_ads/client.py (hunk 2 above)
Comment (BenRKarl) [reply google/ads/google_ads/client.py]: :+1:

File: google/ads/google_ads/client.py (hunk 3 above)
Comment (BenRKarl) [reply google/ads/google_ads/client.py]: :+1:

**Pull request merged**